### PR TITLE
Allow the usage of smallvec 0.4 and bump version

### PR DESCRIPTION
**Pull request by Eijebong**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "parking_lot"
-version = "0.4.4"
+version = "0.4.5"
```
Comment (Amanieu) [inline Cargo.toml]: This isn't necessary, only the `parking_lot_core` version needs to be bumped.

File: core/Cargo.toml [hunk 2]
```diff
@@ -9,7 +9,7 @@ repository = "https://github.com/Amanieu/parking_lot"
 keywords = ["mutex", "condvar", "rwlock", "once", "thread"]
 
 [dependencies]
-smallvec = "0.3"
+smallvec = ">= 0.3, < 0.5"
```
Comment (Amanieu) [inline core/Cargo.toml]: Just use `smallvec = "0.4"`.

**Comment by Amanieu** [comment]
Thanks!

**Comment by Eijebong** [comment]
Thanks for the quick reaction :)

**Pull request merged**